docs/developer-docs/getting-started/deploy-and-manage.mdx

```diff
@@ -0,0 +1,335 @@
+
+- `dfx start`: Returns replica output in the terminal window. A new terminal window will need to be opened to continue running commands.
+
+- `dfx start --background`: Runs the processes in the background and allows you to continue running commands in the current terminal.
+
+
+## Deploy locally 
+
+Deploy the backend canister that you have written code for:
+
+```
+dfx deploy PROJECT_NAME_backend --network=local
+```
+
+:::info
+The `--network=local` flag here is redundant and shown for educational purposes. The local network is the default setting for `dfx` commands; therefore, no flag is needed to target it.
+:::
+
+This will deploy your project to the local replica. In the background, this command:
+
+- Creates the canister(s) specified in the command; if no canister name was specified, it will deploy all canisters listed in the `dfx.json` file.
+
+- Compiles the code for each canister into a Wasm module.
+
+- Installs the Wasm module into each canister.
+
+- Deploys the canister(s) to the deployment endpoint of choice (local, playground, mainnet).
+
+<details>
+<summary>Output</summary>
+
+```
+Deployed canisters.
+URLs:
+  Backend canister via Candid interface:
+    PROJECT_NAME_backend: http://127.0.0.1:4943/?canisterId=be2us-64aaa-aaaaa-qaabq-cai&id=bkyz2-fmaaa-aaaaa-qaaaq-cai
+```
+
+</details>
+
+Your deployed canister will now be executing two timers and printing the output directly to the terminal window:
+
+```
+2024-11-15 19:32:48.584899 UTC: [Canister bkyz2-fmaaa-aaaaa-qaaaq-cai] Generated new random proposal number: 133039
+2024-11-15 19:32:51.197894 UTC: [Canister bkyz2-fmaaa-aaaaa-qaaaq-cai] Proposal info obtained through HTTPS outcall: {"action":"ManageNeuron","action_nns_function":null,"deadline_timestamp_seconds":1726953359,"decided_timestamp_seconds":1726783698,"executed_timestamp_seconds":1726783698,"failed_timestamp_seconds":0,"failure_reason":null,"id":113165,"known_neurons_ballots":[{"id":"28","name":"Internet Computer Association","vote":0},{"id":"27","name":"DFINITY Foundation","vote":0},
+...
+```
+
+You can interact with the canister's public methods either in your web browser through the [Candid UI](/docs/current/developer-docs/smart-contracts/candid/candid-concepts) URL that was returned in the deployment output or using the command line. Candid is an interface description language used to interact with a canister's public methods.
+
+## Upgrades
+
+Canisters can be upgraded, allowing the same canister ID to be retained for an application. Upgrades preserve the state of a canister while allowing you to change the code. However, canister upgrades clear the canister's Wasm memory. Any data not stored in [stable memory](/docs/current/developer-docs/smart-contracts/maintain/storage) will be removed.
+
+In your backend canister source code file, make a change such as increasing the timer to 60 seconds rather than 30 seconds. Save the file.
+
+Then, upgrade your canister to use the edited code:
```

Comment:
before that you should provide instructions to run `dfx build PROJECT_NAME_backend`. this produces a new wasm which will be applied in the command afterwards.